library-best-practices/glossary.md

```diff
@@ -66,3 +66,72 @@ Java Dependency Glossary
 - **Linkage-matchable version** (said of a particular version of A in relation
   to all versions of B): There exists some version of B such that the version of
   A and the version of B are linkage-compatible.
+
+
+### Static Linkage Check
+
+- **Static linkage check**: a process to identify the existence of _static
+  linkage conflicts_ for a given classpath, by scanning Java classes and
+  verifying the availability in the classpath.
+
+- **Class usage graph**: a possibly cyclic directed graph of references between classes
+  in the classpath. The nodes (vertices) of the graph correspond to
+  Java classes and the directed edges are references between classes.
+  For example, when 'Class A' has reference to 'Class B' on calling a method X,
+  the class usage graph holds an edge between two nodes:
+
+  ```
+  [Class A] --(method X of class B)-> [Class B]
+  ```
+
+  In this case, 'Class A' is called the _source class_ of the reference and
+  'Class B' is called the _destination class_.
+
+  In general, there can be multiple (parallel) edges between two nodes when
+  a class is calling two or more methods and fields on another class.
+  Self-loops (references between the same class) are possible and
+  common, however, it is safe to omit such references during static linkage checks,
+  because the self-loop references are ensured to be linkage compatible by Java compiler.
+
+- **A reference**: in the class usage graph, the relationship between two 
+  classes is either _class reference_, _method reference_ or _field reference_.
+
+  A _method reference_ indicates that the source class invokes a (static or
+  non-static) method of the destination class.
+
+  A _field reference_ indicates that the source class accesses a (static or
+  non-static) field of the destination class.
+
+  A _class reference_ indicates that the source class uses the destination
+  class without referencing a specific field or method (e.g., class inheritance).
+
+- **Static linkage error**: a reported error for a reference as the result of
```

Comment:
I feel like this notion belongs in the "Types of conflicts and compatibility" section, although probably with some modification so that it fits. In particular, we need need a general notion of an error beyond just conflicts (the missing case), without any reference to running a tool.